packages/remix-server-runtime/server.ts

```diff
@@ -470,6 +477,32 @@ async function handleDocumentRequest({
   } catch (error: any) {
     responseStatusCode = 500;
 
+    // next up, we'll verify that `meta` returned without error
+    try {
+      await callRouteMeta({
```

Comment:
I wonder if there's any use in distinguishing errors thrown when calling a route `meta()` function here from errors thrown when rendering the `<meta>` tags.  The rendering is far more in our control and we should be able to be pretty confident it won't throw render errors.  The exported `meta` function contents are out of our control - but we _do_ control the execution of it in our own `<Meta>` component.

Spitballing here but what if we did something like this in `<Meta>` 

```js
for (let match of matches) {
  let routeId = match.route.id;
  try {
    let routeMeta = routeModule.meta({ data, parentsData, params, location })
    ...
  } catch (error) {
    // Re-throw the error and associate it with the routeId that threw it
    throw new MetaError(routeId, error);
  }
}
```

Then we could identify the error in this `handleDocumentRequest` `catch` block as a meta-specific error (via `instanceof MetaError`) and know the `routeId` that caused it and look for the nearest ancestor boundary (i.,e., prevent re-calling the problematic `meta` on the second pass - but attempt to call the successful ones above it).

This is effectively what we're doing with the mutable boundaryIds (identifying how deep we got) - but we have no choice there since it's React calling them, not us.  `meta` isn't subject to that same restriction so we could be a bit more direct in the identification of the problematic `routeId`.

Reply:
Actually we'd still need some logic changes in `Meta` to do that too I think.  It doesn't have the concept of `renderableMatches` that we have for `getDocumentHeaders` which I think is what it would need to not call child's meta function on the second pass if we had identified parent as the renderable boundary.